src/libstd/sys/windows/os.rs

```diff
@@ -321,6 +323,9 @@ pub fn args() -> Args {
         let lpCmdLine = c::GetCommandLineW();
         let szArgList = c::CommandLineToArgvW(lpCmdLine, &mut nArgs);
 
+        // szArcList can be NULL if CommandLinToArgvW failed,
+        // but in that case nArgs is 0 so we won't actually
+        // try to read a null pointer
```

Comment:
The docs don't guarantee that nargs is 0 on error, not what I can find. Why not check for error and set nargs to zero in that case?